### ISSUE TITLE
Upgrade from 9.0.1 to 9.0.2 fails

### ISSUE THREAD

**Issue by albertocrj** [owncloud/core#24518]
### Steps to reproduce
1. Install stable9 (9.0.1) tarball
2. Run the updater

### Expected behaviour
Updater upgrades to 9.0.2

### Actual behaviour
Upgrade fails

### Server configuration
Operating system**: Red Hat Enterprise Linux Server release 7.1 (Maipo)

Web server: Apache 2.4.6 + mod_fcgid

Database: mysql 5.5.47

PHP version: PHP 5.6.20

ownCloud version: 9.0.1

Updated from an older ownCloud or fresh install: Fresh install

**Where did you install ownCloud from: Download of owncloud-9.0.1.tar.bz2

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - comments: 0.2
  - dav: 0.1.6
  - documents: 0.12.0
  - encryption: 1.2.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_antivirus: 0.8.0.1
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - user_ldap: 0.8.0
```

**The content of config/config.php:**

```
    "system": {
        "datadirectory": "\/owncloud\/data",
        "trashbin_retention_obligation": 60,
        "enable_previews": true,
        "preview_libreoffice_path": "\/bin\/libreoffice",
        "preview_max_x": 2048,
        "preview_max_scale_factor": 10,
        "preview_max_filesize_image": 50,
        "preview_office_cl_parameters": "",
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown"
        ],
        "instanceid": "ocjl3rwvb6qy",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.local"
        ],
        "overwrite.cli.url": "https:\/\/owncloud.local",
        "dbtype": "mysql",
        "version": "9.0.1.3",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "America\/Sao_Paulo",
        "installed": true,
        "default_language": "pt_BR",
        "defaultapp": "files",
        "session_keepalive": true,
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "ldapIgnoreNamingRules": false,
        "loglevel": "0",
        "singleuser": false,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "127.0.0.1",
            "port": 6379
        },
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }

```

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No 

**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration

```
+-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
| Configuration                 | s01                                                                                                                                     |
+-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 0                                                                                                                                       |
| hasPagedResultSupport         |                                                                                                                                         |
| homeFolderNamingRule          |                                                                                                                                         |
| lastJpegPhotoLookup           | 0                                                                                                                                       |
| ldapAgentName                 | uid=MY_BIND_LOGIN                                                                                        |
| ldapAgentPassword             | ***                                                                                                                                     |
| ldapAttributesForGroupSearch  |                                                                                                                                         |
| ldapAttributesForUserSearch   |                                                                                                                                         |
| ldapBackupHost                |                                                                                                                                         |
| ldapBackupPort                |                                                                                                                                         |
| ldapBase                      | dc=MY_BASE_DN                                                                                                                         |
| ldapBaseGroups                | ou=MY_GROUP_DN                                                                                            |
| ldapBaseUsers                 | ou=MY_USERS_DN                                                                                                                  |
| ldapCacheTTL                  | 600                                                                                                                                     |
| ldapConfigurationActive       | 1                                                                                                                                       |
| ldapDynamicGroupMemberURL     |                                                                                                                                         |
| ldapEmailAttribute            | mail                                                                                                                                    |
| ldapExperiencedAdmin          | 0                                                                                                                                       |
| ldapExpertUUIDGroupAttr       | cn                                                                                                                                      |
| ldapExpertUUIDUserAttr        | uid                                                                                                                                     |
| ldapExpertUsernameAttr        | mail                                                                                                                                    |
| ldapGroupDisplayName          | cn                                                                                                                                      |
| ldapGroupFilter               | (&(objectclass=ntGroup))                                                                                                                |
| ldapGroupFilterGroups         |                                                                                                                                         |
| ldapGroupFilterMode           | 0                                                                                                                                       |
| ldapGroupFilterObjectclass    |                                                                                                                                         |
| ldapGroupMemberAssocAttr      | uniqueMember                                                                                                                            |
| ldapHost                      | ldaps://ldap.local                                                                                                                |
| ldapIgnoreNamingRules         |                                                                                                                                         |
| ldapLoginFilter               | (&(&(objectclass=inetOrgPerson)(memberof=MY_GROUP))(|(uid=%uid)(|(mail=%uid)))) |
| ldapLoginFilterAttributes     |                                                                                                                                         |
| ldapLoginFilterEmail          | 1                                                                                                                                       |
| ldapLoginFilterMode           | 0                                                                                                                                       |
| ldapLoginFilterUsername       | 1                                                                                                                                       |
| ldapNestedGroups              | 1                                                                                                                                       |
| ldapOverrideMainServer        |                                                                                                                                         |
| ldapPagingSize                | 20000                                                                                                                                   |
| ldapPort                      | 636                                                                                                                                     |
| ldapQuotaAttribute            |                                                                                                                                         |
| ldapQuotaDefault              |                                                                                                                                         |
| ldapTLS                       | 0                                                                                                                                       |
| ldapUserDisplayName           | cn                                                                                                                                      |
| ldapUserDisplayName2          | mail                                                                                                                                    |
| ldapUserFilter                | (&(objectclass=inetOrgPerson)(memberof=cn=MY_GROUP))                               |
| ldapUserFilterGroups          |                                                                                                                                         |
| ldapUserFilterMode            | 0                                                                                                                                       |
| ldapUserFilterObjectclass     |                                                                                                                                         |
| ldapUuidGroupAttribute        | auto                                                                                                                                    |
| ldapUuidUserAttribute         | auto                                                                                                                                    |
| turnOffCertCheck              | 0                                                                                                                                       |
| useMemberOfToDetectMembership | 1                                                                                                                                       |
+-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+

```

#### Web server error log
```
[Mon May 09 12:00:06.502493 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: PHP Fatal error:  Uncaught exception 'Symfony\\Component\\Process\\Exception\\ProcessFailedException' with message 'The command "php /var/www/owncloud-9.0.1/occ --no-warnings config:list --private  --output "json"" failed.
[Mon May 09 12:00:06.502529 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: Exit Code: 255(Unknown error)
[Mon May 09 12:00:06.502535 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: Output:
[Mon May 09 12:00:06.502539 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: ================
[Mon May 09 12:00:06.502542 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: Error Output:
[Mon May 09 12:00:06.502546 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: ================
[Mon May 09 12:00:06.502549 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: PHP Fatal error:  Class OCA\\Encryption\\Crypto\\Encryption contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (OCP\\Encryption\\IEncryptionModule::isReadyForUser) in /var/www/owncloud-9.0.1/apps/encryption/lib/crypto/encryption.php on line 550
[Mon May 09 12:00:06.502553 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: ' in /var/www/owncloud-9.0.1/updater/src/Utils/OccRunner.php:50
[Mon May 09 12:00:06.502557 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: Stack trace:
[Mon May 09 12:00:06.502560 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: #0 /var/www/owncloud-9.0.1/updater/src/Utils/OccRunner.php(56): Owncloud\\Updater\\Utils\\OccRunner->run('--no-warnings c...')
[Mon May 09 12:00:06.502572 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: #1 /var/www/owncloud-9.0.1/updater/src/Utils/ConfigReader.php(106): Owncloud\\Updater\\Utils\\OccRunner->runJson('config:list --p...')
[Mon May 09 12:00:06.502576 2016] [fcgid:warn] [pid 2785] [client MY_CLIENT_IP:54441] mod_fcgid: stderr: #2 /var/www/owncloud-9.0.1/updater/src/Utils/ConfigReader.php(50): Owncloud\\Updater\\Utils\\ConfigReader->load() in /var/www/owncloud-9.0.1/updater/src/Utils/OccRunner.php on line 50

```

#### Updater log (update.log)
```
[info] Execution of upgrade:checkpoint command started
[info] Execution of upgrade:checkpoint command stopped. Exit code is 0
[info] Execution of upgrade:detect command started
[info] ownCloud 9.0.1.3 found
[info] Execution of upgrade:detect command stopped. Exit code is 0
[info] Execution of upgrade:checkSystem command started
[info] Execution of upgrade:checkSystem command stopped. Exit code is 0
[info] Execution of upgrade:checkpoint command started
[info] Execution of upgrade:checkpoint command stopped. Exit code is 0
[info] Execution of upgrade:detect command started
[info] ownCloud 9.0.1.3 found
[info] Execution of upgrade:detect command stopped. Exit code is 0
[info] Execution of upgrade:disableNotShippedApps command started
[info] Execution of upgrade:disableNotShippedApps command stopped. Exit code is 0
[info] Execution of upgrade:executeCoreUpgradeScripts command started
[debug] Moving config/config.sample.php
[debug] Moving core
[debug] Moving l10n
[debug] Moving lib
[debug] Moving ocs
[debug] Moving ocs-provider
[debug] Moving resources
[debug] Moving settings
[debug] Moving .htaccess
[debug] Moving .mailmap
[debug] Moving .tag
[debug] Moving .user.ini
[debug] Moving AUTHORS
[debug] Moving console.php
[debug] Moving COPYING-AGPL
[debug] Moving cron.php
[debug] Moving db_structure.xml
[debug] Moving index.html
[debug] Moving index.php
[debug] Moving indie.json
[debug] Moving occ
[debug] Moving public.php
[debug] Moving remote.php
[debug] Moving robots.txt
[debug] Moving status.php
[debug] Moving version.php
[info] Execution of upgrade:executeCoreUpgradeScripts command stopped. Exit code is 0

```

**Comment by mundschenk-at** [comment]
Same problem on FreeBSD with nginx.

**Comment by leonbroseliske** [comment]
Same issue here with an upgrade 9.0.3.1 -> 9.0.2.2 on Debian Jessie / Apache 2.4.10 / PHP 5.6.20 / MariaDB 10.1.13

**Comment by RealRancor** [comment]
I think updates with the updater app should be reported to: https://github.com/owncloud/updater/issues

@mundschenk-at @leonbroseliske Before posting a "same problem" or "same issue" make sure it is indeed the same issue. Are you also getting the errors posted above or is your update "just failing"?

**Comment by albertocrj** [comment]
@RealRancor Sorry, I didnt release that. Do I need to repost on updater issues?

**Comment by DeepDiver1975** [comment]
This has to go into the updater repo.
It is related to the issue that the updater is updating apps step by step instead of all core/shipped apps together. :angry:

**Comment by albertocrj** [comment]
reposted on updater issues -> https://github.com/owncloud/updater/issues/350

Thanks

**Comment by ET-Bent** [comment]
Is there any way to quick-fix this issue as downgrading is not supported?

**Comment by mundschenk-at** [comment]
@RealRancor: I did indeed get the exact same error messages (except for a slightly different location of my owncloud directory of course) - otherwise I would have created a separate issue report. I concede that I should have checked for the updater component, though.

AS @DeepDiver1975 has hinted, it looks like the problem is that the base interfaces were upgraded, but not the encryption app, leading to a fatal error preventing the updater from continuing the process.

@ET-Bent I fixed my installation by doing a [manual update from the shell](https://doc.owncloud.org/server/9.0/admin_manual/maintenance/manual_upgrade.html).
